pkg/restore/crd.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rook/kubectl-rook-ceph/pkg/debug"
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func RestoreCrd(ctx context.Context, k8sclientset *k8sutil.Clientsets, operatorNamespace, clusterNamespace string, args []string) {
+	crd := args[0]
+	logging.Info("Detecting which ceph crd resources to restore")
+	getCrName := `kubectl -n %s get %s -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.deletionGracePeriodSeconds}{"\n"}{end}' | awk '$2=="0" {print $1}'`
+	command := fmt.Sprintf(getCrName, clusterNamespace, crd)
+	crName := strings.TrimSpace(exec.ExecuteBashCommand(command))
```

Comment:
What if there are multiple deleted resources? We wouldn't expect this for a CephCluster, but it could happen for other CRs.

Reply:
Good question, I'll need to test this but currently, it will get all the names. It requires in command

Reply:
Get a chance to look at this?

Reply:
somehow it was skipped with all the comments, I'll look this tomorrow

Reply:
Good catch, I need to add `| head -n 1`  in the end to get only the first line.

Reply:
it is working now

Reply:
So if there are multiple deleted resources and they don't specify the CR name, they need to run the command again to restore all of them, right? Maybe mention that in the doc.